clarity_ext/domain/container.py

```diff
@@ -180,8 +187,16 @@ def set_well(self, well_pos, artifact_name=None, artifact_id=None, artifact=None
         # TODO: Accept only an artifact
         self.wells[well_pos].artifact = artifact
 
+    def __iter__(self):
```

Comment:
Are you sure this works as expected? I'm not 100% what is the expected behavior  here but won't this get you a generator of a generator instead of looping over the initial generator?

Here is a small ipython experiment that might show my point:
```
In [1]: def inner_yield():
   ...:     yield 'a'
   ...:     

In [2]: def outer_yield():
   ...:     yield inner_yield()
   ...:     

In [3]: outer_yield()
Out[3]: <generator object outer_yield at 0x7f549103c320>

In [4]: list(outer_yield())
Out[4]: [<generator object inner_yield at 0x7f5491051af0>]

In [5]: list(list(outer_yield()))
Out[5]: [<generator object inner_yield at 0x7f5490f1fc80>]

In [6]: def outer_yield():
   ...:     return inner_yield()
   ...: 

In [7]: outer_yield()
Out[7]: <generator object inner_yield at 0x7f5491051aa0>

In [8]: list(outer_yield())
Out[8]: ['a']
```

So maybe I'm misunderstanding something here because I don't fully understand the domain, but I'd thought that you could just do `return self.enumerate_wells(order=self.DOWN_FIRST)`